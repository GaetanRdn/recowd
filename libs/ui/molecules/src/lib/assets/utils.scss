@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
