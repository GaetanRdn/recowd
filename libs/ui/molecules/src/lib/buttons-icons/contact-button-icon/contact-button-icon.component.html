<a rcButton role="button">
  <rc-material-icon type="outlined">mail</rc-material-icon>
</a>
