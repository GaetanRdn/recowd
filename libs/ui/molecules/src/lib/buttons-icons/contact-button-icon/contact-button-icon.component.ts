import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ButtonDirective, MaterialIconDirective } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-contact-button-icon',
  standalone: true,
  imports: [ButtonDirective, MaterialIconDirective],
  templateUrl: './contact-button-icon.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ContactButtonIconComponent {}
