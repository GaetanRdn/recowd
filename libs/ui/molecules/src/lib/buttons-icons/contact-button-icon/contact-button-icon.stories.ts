import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { ContactButtonIconComponent } from './contact-button-icon.component';

export default {
  title: 'molecules/buttons-icons/contact-button-icon',
  decorators: [
    moduleMetadata({
      imports: [ContactButtonIconComponent],
    }),
  ],
} as Meta<ContactButtonIconComponent>;

const Template: Story<ContactButtonIconComponent> = () => ({
  template: `<rc-contact-button-icon></rc-contact-button-icon>`,
});

export const Default = Template.bind({});
