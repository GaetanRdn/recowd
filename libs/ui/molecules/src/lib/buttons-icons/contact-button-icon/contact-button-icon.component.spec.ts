import { TestBed, waitForAsync } from '@angular/core/testing';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';
import { ContactButtonIconComponent } from './contact-button-icon.component';

describe('ContactButtonIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [ContactButtonIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});

@Component({
  template: ` <rc-contact-button-icon></rc-contact-button-icon>`,
})
export class HostComponent {}
