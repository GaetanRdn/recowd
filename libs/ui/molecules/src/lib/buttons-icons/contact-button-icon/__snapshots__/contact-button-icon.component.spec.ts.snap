// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactButtonIconComponent should create 1`] = `
<rc-contact-button-icon>
  <a
    class="rc-button"
    rcbutton=""
    role="button"
  >
    <rc-material-icon
      class="material-icons-outlined"
      type="outlined"
    >
      mail
    </rc-material-icon>
  </a>
</rc-contact-button-icon>
`;
